lib/fhir_client/client.rb

```diff
@@ -39,6 +56,7 @@ class Client
     # @return
     #
     def initialize(base_service_url, default_format: FHIR::Formats::ResourceFormat::RESOURCE_JSON, proxy: nil)
+      base_service_url = "http://#{base_service_url}" unless base_service_url.start_with?("https://", "http://")
```

Comment:
Instead of dealing with these URLS as strings to be manipulated and mutated, what do you think about using the ruby URI object?

```ruby
uri = URI.parse(base_service_url)

# Check if the scheme is missing
if uri.scheme.nil?
  # Construct a new URI with the "http" scheme
  uri = URI::HTTP.build(host: uri.host, path: uri.path)
end
@base_service_url = uri
```